server/sonar-server/src/main/java/org/sonar/server/es/IndexDefinitions.java

```diff
@@ -101,12 +102,15 @@ public IndexDefinitions(IndexDefinition[] defs) {
   public void start() {
     // collect definitions
     IndexDefinition.IndexDefinitionContext context = new IndexDefinition.IndexDefinitionContext();
-    for (IndexDefinition definition : defs) {
-      definition.define(context);
-    }
 
-    for (Map.Entry<String, NewIndex> entry : context.getIndices().entrySet()) {
-      byKey.put(entry.getKey(), new Index(entry.getValue()));
+    if (!settings.getBoolean("sonar.internal.es.disableIndexes")) {
```

Comment:
this makes the `IndexDefinitions` empty which in turn makes sure no index is created

I like the fact that if there should be no index created into ES anyway, then there is no `IndexDefinition` in this class too